### PR TITLE
add PredictKube Scaler docs

### PR DESCRIPTION
**Pull request by daniel-yavorovich**
This PR adds documentation for a new PredictKube Scaler.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Relates to https://github.com/kedacore/keda/pull/2418 & https://github.com/kedacore/keda/issues/2458

### CODE REVIEW

File: content/docs/2.6/scalers/predictkube.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+- `queryStep` - The maximum time between two slices within the boundaries for QML range query, used in the query.
+- `threshold` - Value to start scaling for.
+
+### Authentication Parameters
+
+Predictkube Scaler supports one type of authentication - authentication by API key.
+
+**Auth gateway based authentication:**
+
+- `apiKey` - API key previously issued for this tenant.
```
Comment (tomkerkhove) [inline content/docs/2.6/scalers/predictkube.md:44]: API key of what? :)

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
It looks like PredictKube is a service people need an API key for. I think it would be good if:
1. There is an introduction added before `Trigger Specification` clarifying that PredictKube is a service
2. Provide transparency on the data usage that is being sent to PredictKube. This can be as simple as adding a link to a privacy/legal page on your official docs but it should provide transparency if customer metrics are used to train the model as well

Thank you!

**Comment by daniel-yavorovich** [comment]
> It looks like PredictKube is a service people need an API key for. I think it would be good if:
> 
> 1. There is an introduction added before `Trigger Specification` clarifying that PredictKube is a service
> 2. Provide transparency on the data usage that is being sent to PredictKube. This can be as simple as adding a link to a privacy/legal page on your official docs but it should provide transparency if customer metrics are used to train the model as well
> 
> Thank you!

Thanks for this important point, Tom! We're currently working on our full policy to publish it in the nearest future.

**Comment by tomkerkhove** [comment]
Thank you! But I'd say this is a mandatory thing to have before we can merge this feature as it is pretty important for end-users to understand.

File: content/docs/2.6/scalers/predictkube.md (hunk 1 above)
Comment (daniel-yavorovich) [reply content/docs/2.6/scalers/predictkube.md:44]: for PredictKube API. It can be obtained for free at https://predictkube.com/

Comment (tomkerkhove) [reply content/docs/2.6/scalers/predictkube.md:44]: By now I've figured that out, but this is something that should be as part of the documentation.

Can you add that please?

File: content/docs/2.6/scalers/predictkube.md [hunk 2]
```diff
@@ -7,6 +7,12 @@ layout = "scaler"
 go_file = "predictkube_scaler"
 +++
 
+PredictKube as an open-source project with the SAAS part consisting of an AI model that requires API to connect to the project for the main functions of predicting and scaling.
+
+To make our AI model access your data and make a prediction based on it, please use this API key we provide below.
```
Comment (tomkerkhove) [inline content/docs/2.6/scalers/predictkube.md:12]: This feels more accurate given it is not issued by KEDA, what do you think?
```suggestion
To make our AI model access your data and make a prediction based on it, please use this API key we provide in PredictKube.
```

File: content/docs/2.6/scalers/predictkube.md [hunk 3]
```diff
@@ -7,6 +7,12 @@ layout = "scaler"
...
+
+Review our [Privacy Policy](https://predictkube.com/privacy-policy) to see how your data circulates inside PredictKube.
```
Comment (tomkerkhove) [inline content/docs/2.6/scalers/predictkube.md:14]: As per your policy page:
```suggestion
Review our [Privacy Policy](https://predictkube.com/privacy-policy) to see how your data circulates in and out PredictKube.
```

File: content/docs/2.6/scalers/predictkube.md (hunk 1 above)
Comment (daniel-yavorovich) [reply content/docs/2.6/scalers/predictkube.md]: Sure, we have added comment in https://github.com/dysnix/keda-docs/commit/1ff8f1dc02114b98a7c301f7025e05c33830ecb7

Thank you!

File: content/docs/2.6/scalers/predictkube.md [hunk 4]
```diff
@@ -7,6 +7,12 @@ layout = "scaler"
 go_file = "predictkube_scaler"
 +++
 
+PredictKube as an open-source project with the SAAS part consisting of an AI model that requires API to connect to the project for the main functions of predicting and scaling.
```
Comment (tomkerkhove) [inline content/docs/2.6/scalers/predictkube.md:10]: How about this?
```suggestion
PredictKube is an open-source project that uses a SaaS offering consisting of an AI model consumed by connecting to a REST API for the main functions of predicting and scaling.
```

File: content/docs/2.6/scalers/predictkube.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
++++
+title = "Predictkube"
+availability = "v1.0+"
```
Comment (tomkerkhove) [inline content/docs/2.6/scalers/predictkube.md:3]:
```suggestion
availability = "v2.6+"
```

File: content/docs/2.6/scalers/predictkube.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+maintainer = "Dysnix"
+description = "AI-based predictive scaling based on Prometheus metrics."
```
Comment (tomkerkhove) [inline content/docs/2.6/scalers/predictkube.md:5]:
```suggestion
description = "AI-based predictive scaling based on Prometheus metrics & PredictKube SaaS."
```

**Review by tomkerkhove — APPROVED** [review]
LGTM, but waiting for feature to be merged.

File: content/docs/2.6/scalers/predictkube.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+- `predictHorizon` - Prediction time interval. It is usually equal to the cool-down period of your application.
+- `historyTimeWindow` - Time range for which to request metrics from Prometheus. We recommend using minimum 7-14 days time window as historical data.
+- `prometheusAddress` - Address of Prometheus server.
+- `metricName` - Name to identify the Metric in the external.metrics.k8s.io API. API. If using more than one trigger it is required that all `metricName`(s) be unique.
+- `query` - Predict the query that will yield the value for the scaler to compare against the `threshold`. The query must return a vector/scalar single element response.
+- `queryStep` - The maximum time between two slices within the boundaries for QML range query, used in the query.
+- `threshold` - Value to start scaling for.
+
+### Authentication Parameters
```
Comment (zroubalik) [inline content/docs/2.6/scalers/predictkube.md:44]: Would be great to mention, that Authentication for referenced Prometheus works as well. Noting it here in this section and in the example Secret

File: content/docs/2.6/scalers/predictkube.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    queryStep: "2m"
+    threshold: '100'
+```
+
+**Parameter list:**
+
+- `predictHorizon` - Prediction time interval. It is usually equal to the cool-down period of your application.
+- `historyTimeWindow` - Time range for which to request metrics from Prometheus. We recommend using minimum 7-14 days time window as historical data.
+- `prometheusAddress` - Address of Prometheus server.
+- `metricName` - Name to identify the Metric in the external.metrics.k8s.io API. API. If using more than one trigger it is required that all `metricName`(s) be unique.
```
Comment (zroubalik) [inline content/docs/2.6/scalers/predictkube.md:39]: As mentioned on the main PR, we are trying to not expose this property for users, unless there is a really good reason.

File: content/docs/2.6/scalers/predictkube.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+- `query` - Predict the query that will yield the value for the scaler to compare against the `threshold`. The query must return a vector/scalar single element response.
+- `queryStep` - The maximum time between two slices within the boundaries for QML range query, used in the query.
```
Comment (tomkerkhove) [inline content/docs/2.6/scalers/predictkube.md:41]: What about this?
```suggestion
- `queryWindow` - The maximum time between two slices within the boundaries for QML range query, used in the query.
```

File: content/docs/2.6/scalers/predictkube.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+- `historyTimeWindow` - Time range for which to request metrics from Prometheus. We recommend using minimum 7-14 days time window as historical data.
+- `prometheusAddress` - Address of Prometheus server.
+- `query` - Predict the query that will yield the value for the scaler to compare against the `threshold`. The query must return a vector/scalar single element response.
+- `queryStep` - The maximum time between two slices within the boundaries for QML range query, used in the query.
+- `threshold` - Value to start scaling for.
+
+### Authentication Parameters
+
+Predictkube Scaler supports one type of authentication - authentication by API key.
+Prometheus supports all authentication methods that are available in Prometheus Scaler.
```
Comment (zroubalik) [inline content/docs/2.6/scalers/predictkube.md:45]:
```suggestion
Prometheus used in Predictkube Scaler supports all authentication methods that are available in Prometheus Scaler.
```

File: content/docs/2.6/scalers/predictkube.md (hunk 9 above)
Comment (zroubalik) [reply content/docs/2.6/scalers/predictkube.md:39]: I am fine with both 🤷

**Review by zroubalik — APPROVED** [review]

**Comment by zroubalik** [comment]
@daniel-yavorovich looking good, could you please properly rebase this PR, so it contains only PredictKube related commits? Then we are good to go

**Comment by daniel-yavorovich** [comment]
> looking good, could you please properly rebase this PR, so it contains only PredictKube related commits? Then we are good to go

done, thank you

**Comment by zroubalik** [comment]
> > looking good, could you please properly rebase this PR, so it contains only PredictKube related commits? Then we are good to go
> 
> done, thank you

@daniel-yavorovich I can see 32 commits in this PR and majority of them non related to this PR

**Review by tomkerkhove — APPROVED** [review]

**Pull request merged**